tests/php/FileTest.php

```diff
@@ -999,4 +999,41 @@ public function testCanEditFallsBackToCheckingDefaultFilePermissions()
         $file->ID = 234;
         $this->assertFalse($file->canEdit());
     }
+
+    public function testHasRestrictedAccess()
```

Comment:
This test looks like a prime candidate to use a [data provider](https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers)

Reply:
(I've never used dataProvider before)

I'm open-minded to refactoring to this, though
- Would all the DataObjects be created properly and all the Parent() associations as well?
- Do we actually want to encourage two ways of doing something instead of just using one? i.e. Creating DataObjects yml fixtures (de facto standard) vs dataProvider (new and maybe better standard)?

Reply:
You can't really scaffold anything dynamic in dataproviders well. I think here Dan's suggestion would be to pass the fixture name and the expectation:

```php
public function restrictedAccessProvider()
{
    yield ['restricted-test-r', false];
    yield ['restricted-test-r1', false];
    yield ['restricted-test-r11', false];
}

/**
 * @dataProvider restrictedAccessProvider
 */
public function testHasRestrictedAccess(string $fixtureName, bool $expected)
{
    $fixture = $this->objFromFixture(Folder::class, $fixtureName);
    $this->assertSame($expected, $fixture->hasRestrictedAccess());
}
```

etc

Reply:
cool, have implemented